### PR TITLE
feat(cli/lib): use scene background color as default

### LINKED ISSUE #130: [FEATURE] read background color from `manim.config`, save it in config and use it in Manim-Slides
### Description

As mentioned in https://github.com/jeertmans/manim-slides/issues/128#issuecomment-1443429114, it would be nice to use the save background color as the one used with Manim (or ManimGL).

A few to-dos:
- [ ] Add a new `background_color` field in https://github.com/jeertmans/manim-slides/blob/a9f53555951b490a74f00acff70c2820c293a8f9/manim_slides/config.py#L147-L149
- [ ] Read `background_color` from Manim / ManimGL config and write it somewhere in https://github.com/jeertmans/manim-slides/blob/a9f53555951b490a74f00acff70c2820c293a8f9/manim_slides/slide.py#L28-L54
- [ ] Default to presentation background color if `--background-color`/`-cbackground-color` option is not set in **present** https://github.com/jeertmans/manim-slides/blob/a9f53555951b490a74f00acff70c2820c293a8f9/manim_slides/present.py#L783-L792 and in **convert** https://github.com/jeertmans/manim-slides/blob/a9f53555951b490a74f00acff70c2820c293a8f9/manim_slides/convert.py#L271

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Fairlight8**
## Fixes Issue

Closes #130 

## Check List (Check all the applicable boxes)

- [x] I understand that my contributions needs to pass the checks.
- [x] If I created new functions / methods, I documented them and add type hints.
- [x] If I modified already existing code, I updated the documentation accordingly.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers
Please review the code, just in case ^^

### CODE REVIEW

File: manim_slides/present.py
Comment (jeertmans) [inline manim_slides/present.py:739]:
```suggestion
        self.label.setStyleSheet(
            f"background-color: {self.thread.current_background_color}"
        )
```

For readability, create a `current_background_color` property :-)

File: manim_slides/slide.py
Comment (jeertmans) [inline manim_slides/slide.py:56]: This does not work because the `background_color` attribute is not placed in `config` with ManimGL. So you need to a separate implementation, as with `__resolution`:

https://github.com/jeertmans/manim-slides/blob/2a327c470bcea5821f4ff21cf1085a81587b745c/manim_slides/slide.py#L57-L63

**Review by jeertmans — CHANGES_REQUESTED** [review]
Thank you for your work @Fairlight8, I requested a few changes before I can merge :-)

File: manim_slides/slide.py
Comment (Fairlight8) [reply manim_slides/slide.py:56]: Oh, I Didn't know, understood!

**Comment by Fairlight8** [comment]
Of course, of course. Sorry for the potential mistakes. It is a simple issue but I am still new contributing to other projects.

I agree with all the changes. Do I have to accept them?

**Comment by jeertmans** [comment]
No problem, first contributions are always hard!

In general, you can approve them, yes. But here, some codes changes are still needed beyond those I proposed. For example, you need to import the Color from pydantic first :)

**Comment by Fairlight8** [comment]
Ohhhh, true true true. So, what's the process now? Should I add a different PR? Add new commits? Can I change this PR for another commit?

**Comment by jeertmans** [comment]
You can just push new commits :)

**Comment by Fairlight8** [comment]
Added some changes! I think  that's fine now!

File: manim_slides/slide.py
Comment (jeertmans) [inline manim_slides/slide.py:58]:
```suggestion
    def __background_color(self) -> str:
```

The return type is not `Tuple[int, int]` :-)

File: manim_slides/slide.py
Comment (jeertmans) [inline manim_slides/slide.py:63]:
```suggestion
        if MANIMGL:
            return self.camera_config["background_color"].hex
        else:
            return config["background_color"].hex
```

File: manim_slides/slide.py
Comment (jeertmans) [inline manim_slides/slide.py:335]:
```suggestion
                    background_color=self.__background_color,
```

**Review by jeertmans — CHANGES_REQUESTED** [review]

**Comment by jeertmans** [comment]
Nice work! A few last changes before merging this were requested. You can see that `pre-commit.ci` does some type checking, and a few errors were pointed-out: https://results.pre-commit.ci/run/github/513032928/1679445219.3lZ-NZzDS-2nlcoJtXckeQ.

You can just approve the changes if you agree with them.

File: manim_slides/slide.py
Comment (Fairlight8) [reply manim_slides/slide.py:58]: True, I forgot the copypaste. Shouldn't the return hint be pydantic.color.Color?

Comment (Fairlight8) [reply manim_slides/slide.py:58]: Okay, nevermind, I just realized that you proposed to return .hex directly inside the property. Gotit!

**Comment by Fairlight8** [comment]
Another commit for those details. By the way. The github pages thing is my issue,  right?

File: manim_slides/slide.py
Comment (jeertmans) [inline manim_slides/slide.py:63]:
```suggestion
            return self.camera_config["background_color"].hex  # typing: ignore
        else:
            return config["background_color"].hex  # typing: ignore
```

**Review by jeertmans — CHANGES_REQUESTED** [review]

**Comment by jeertmans** [comment]
Yep sorry, I didn't want to bother you, so I committed the changes directly ^^

Looks good to me, I will merge this and soon release this :-) Probably as part of 4.12

**Comment by Fairlight8** [comment]
Understood, thank you! Again, thanks a lot  for your patience. Probably, it would be much more shorter and easier if you did that on your own, so I appreciate the patience while I learn all these things!

**Pull request merged**
